components/BottomNavigation/tests/unit/Theming/MDCBottomNavigationThemingTests.m

```diff
@@ -0,0 +1,180 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MaterialBottomNavigation+Theming.h"
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialColorScheme.h"
+#import "MaterialContainerScheme.h"
+#import "MaterialShadowElevations.h"
+#import "MaterialTypography.h"
+#import "MaterialTypographyScheme.h"
+
+static const CGFloat kUnselectedPrimaryAlpha = 0.74f;
+static const CGFloat kUnselectedSurfaceAlpha = 0.6f;
+
+@interface MDCBottomNavigationThemingTests : XCTestCase
+@property(nonatomic, strong, nullable) MDCBottomNavigationBar *bottomNavigationBar;
+@end
+
+@implementation MDCBottomNavigationThemingTests
+
+- (void)setUp {
+  [super setUp];
+
+  self.bottomNavigationBar = [[MDCBottomNavigationBar alloc] init];
+  UITabBarItem *tabBarItem1 = [[UITabBarItem alloc] initWithTitle:@"Home"
+                                                            image:[UIImage imageNamed:@"Home"]
+                                                              tag:0];
+  UITabBarItem *tabBarItem2 = [[UITabBarItem alloc] initWithTitle:@"Messages"
+                                                            image:[UIImage imageNamed:@"Email"]
+                                                              tag:0];
+  UITabBarItem *tabBarItem3 = [[UITabBarItem alloc] initWithTitle:@"Favorites"
+                                                            image:[UIImage imageNamed:@"Favorite"]
+                                                              tag:0];
+  self.bottomNavigationBar.items = @[ tabBarItem1, tabBarItem2, tabBarItem3 ];
+}
+
+- (void)tearDown {
+  self.bottomNavigationBar = nil;
+
+  [super tearDown];
+}
+
+- (void)testPrimaryThemeWithDefaultContainerScheme {
+  // Given
+  MDCBottomNavigationBar *bottomNav = self.bottomNavigationBar;
+  MDCContainerScheme *containerScheme = [[MDCContainerScheme alloc] init];
+
+  // When
+  [bottomNav applyPrimaryThemeWithScheme:containerScheme];
+
+  // Then
+  [self helperTextPrimaryThemeColorForColorScheme:containerScheme.colorScheme];
+  [self helperTestItemTitleFontEqualsFont:containerScheme.typographyScheme.caption];
+  [self helperTestNonsubsytemValuesForBottomNavigationBar:bottomNav];
+}
+
+- (void)testPrimaryThemeWithCustomContainerScheme {
+  // Given
+  MDCBottomNavigationBar *bottomNav = self.bottomNavigationBar;
+  MDCContainerScheme *containerScheme = [[MDCContainerScheme alloc] init];
+  containerScheme.typographyScheme = [self customTypographyScheme];
+  containerScheme.colorScheme = [self customColorScheme];
+
+  // When
+  [bottomNav applyPrimaryThemeWithScheme:containerScheme];
+
+  // Then
+  [self helperTextPrimaryThemeColorForColorScheme:containerScheme.colorScheme];
+  [self helperTestItemTitleFontEqualsFont:containerScheme.typographyScheme.caption];
+  [self helperTestNonsubsytemValuesForBottomNavigationBar:bottomNav];
+}
+
+- (void)helperTextPrimaryThemeColorForColorScheme:(id<MDCColorScheming>)colorScheme {
```

Comment:
Is "helperText" a typo? Did you mean "helperTest"?  I would suggest naming something like "assertPrimaryTheme..." since this method contains asserts.

Reply:
Done